paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -49,518 +46,589 @@
+            with Timeout(seconds=drain_timeout + 300):
+                while True:
+                    instance_id = slave['instance_id']
+                    if not instance_id:
+                        log.warning("Didn't find instance ID for slave: {0}. Skipping terminating".format(slave['pid']))
+                        continue
+                    # Check if no tasks are running or we have reached the maintenance window
+                    if is_safe_to_kill(slave['hostname']) or dry_run:
+                        log.info("TERMINATING: {0} (Hostname = {1}, IP = {2})".format(
+                            instance_id,
+                            slave['hostname'],
+                            slave['ip'],
+                        ))
+                        try:
+                            ec2_client.terminate_instances(InstanceIds=[instance_id], DryRun=dry_run)
+                        except ClientError as e:
+                            if e.response['Error'].get('Code') == 'DryRunOperation':
+                                pass
+                            else:
+                                raise
+                        break
+                    else:
+                        log.info("Instance {0}: NOT ready to kill".format(instance_id))
+                    log.debug("Waiting 5 seconds and then checking again")
+                    time.sleep(5)
+        except TimeoutError:
+            log.error("Timed out after {0} waiting to drain {1}, now terminating anyway".format(drain_timeout,
+                                                                                                slave['pid']))
+            try:
+                ec2_client.terminate_instances(InstanceIds=instance_id, DryRun=dry_run)
+            except ClientError as e:
+                if e.response['Error'].get('Code') == 'DryRunOperation':
+                    pass
                 else:
-                    log.info("Instance {0}: NOT ready to kill".format(instance_id))
-                log.debug("Waiting 5 seconds and then checking again")
-                time.sleep(5)
-    except TimeoutError:
-        log.error("Timed out after {0} waiting to drain {1}, now terminating anyway".format(drain_timeout,
-                                                                                            slave['pid']))
+                    raise
+
+    def sort_slaves_to_kill(self, slaves):
+        """Pick the best slaves to kill. This returns a list of slaves
+        after sorting in preference of which slaves we kill first.
+        It sorts first by number of chronos tasks, then by total number of tasks
+
+        :param slaves: list of slaves dict
+        :returns: list of slaves dicts"""
+        return sorted(slaves, key=lambda x: (x['task_counts'].chronos_count, x['task_counts'].count))
+
+    def scale_resource(self, current_capacity, target_capacity):
+        """Scales a spot fleet request by delta to reach target capacity
```

Comment:
this docs don't seem correct. Can you confirm that there is nothing spotfleet specific in here?